### PR TITLE
fix: export getDefaultPlugins and getKeyboardShortcuts methods

### LINKED ISSUE #373: [BUG]: Default plugin order prevents some custom keymaps
# I'm submitting a

- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [X] Bug report
- [ ] Enhancement
- [X] Feature request
- [ ] Documentation issue or request

## Description

For reference: https://github.com/sibiraj-s/ngx-editor/blob/e6a080d6c3e498a6b56801774ca0cc892a859c70/projects/ngx-editor/src/lib/Editor.ts#L117

The default plugins are added to the plugins array before the plugins provided in options. This works fine in most non-conflicting cases, but prevents creating custom keymaps for i.e. the Enter key, or using other ProseMirror plugins that use such keymaps. The default plugins add keymap bindings earlier in the pipeline that return `true` preventing further keymaps from being invoked. As far as I can tell, the only way around this currently is to set `keyboardShortcuts: false` in the options, and re-create the default keymap plugin by copying the `getKeyboardShortcuts` code (since that function is not exported). I've tried editing the views plugin array after creation and that breaks things without giving an error to the console. Copying the `getKeyboardShortcuts` method works okay, but then you lose the ability to benefit from community improvements and bugfixes to that code with npm updates.

I wasn't quite sure whether to call this a bug report or a feature request, as it's somewhat both. As this excellent library was thoughtfully designed to be flexible and extensible, I would imagine that not being able to override keymaps by default would be an unintended bug. But at the same time, changing the order to put the default plugins last would be a breaking change, so I'm proposing a new config option to work around this issue.

I'd be happy to submit a PR with whichever solution below you feel is best, or if you have any other ideas on how to solve this problem.

## Proposed Solution

Changing the order to put default plugins last (so that custom keymaps are evaluated first) would possibly be a breaking change, although if you wanted to go that route that would be fine by me.

Instead, I'd like to propose a new Options interface property of `preDefaultPlugins?: Plugin[];`. (Naming things is hard, I'm open to other name suggestions.) The default value for this new property in the `DEFAULT_OPTIONS` object would be an empty array.

A new line would be added below https://github.com/sibiraj-s/ngx-editor/blob/e6a080d6c3e498a6b56801774ca0cc892a859c70/projects/ngx-editor/src/lib/Editor.ts#L104:

```ts
const preDefaultPlugins: Plugin[] = options.preDefaultPlugins ?? [];
```

And then the line up above spreading the plugin array would become:
```ts
plugins: [...preDefaultPlugins, ...defaultPlugins, ...plugins],
```

## Alternate Solution 1

Export the `getDefaultPlugins`, `getKeyboardShortcuts`, etc methods in the public API via `public_api.ts`. This would allow users that need custom keymaps to come first to set `keyboardShortcuts: false`, call `getKeyboardShortcuts` themselves, and add that to the array manually in whatever order they need. I think I'd prefer my proposed solution, as that doesn't have to disable and then re-enable defaults.

## Alternate Solution 2

Allow for providing a callback function to build the plugins array. For example:

```ts
export type PluginsBuilder = (defaultPlugins: Plugin[]) => Plugin[];

// add to Options interface: pluginsBuilder?: PluginsBuilder

// in Editor.createEditor, something like:

    const plugins: Plugin[] = options.plugins ?? [];
    const pluginsBuilder: PluginsBuilder = options.pluginsBuilder ?? p => p;
    const attributes: Record<string, string> = options.attributes ?? {};

    const defaultPlugins = [...getDefaultPlugins(schema, {
      history,
      keyboardShortcuts,
      inputRules
    }), ...plugins];

    this.view = new EditorView(null, {
      state: EditorState.create({
        doc,
        schema,
        plugins: pluginsBuilder(defaultPlugins),
      }),
      nodeViews,
      dispatchTransaction: this.handleTransactions.bind(this),
      attributes
    });

```

### Version Information

```
ngx-editor: 12.0.0
angular: 13.0.3
```

### Browser

N/A

### PR DESCRIPTION
**Pull request by paulirwin**
In order to submit this PR, I had to add support for building on Windows too using `run-script-os`.

This PR contains:

- [X] bugfix
- [ ] feature
- [ ] refactor
- [ ] tests
- [ ] documentation
- [ ] others

### Breaking Changes?

- [ ] yes
- [X] no

### Checklist

- [X] commit messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
  - A document related commit is prefixed "docs:"
- [ ] docs have been added / updated (for bug fixes / features)

TODO: is a docs update needed here?

### Does this PR affects any existing issues?

- [X] yes
- [ ] no

Fixes #373

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,9 @@
     "test:demo": "ng test demo --browsers ChromeHeadless --watch false",
     "test:lib": "ng test ngx-editor --browsers ChromeHeadless --watch false",
     "test": "npm run test:lib && npm run test:demo",
-    "prepare": "sh ./scripts/install-husky.sh"
+    "prepare": "run-script-os",
```
Comment (sibiraj-s) [inline package.json:19]:
```bash
npm i -D is-ci
```

we can install `is-ci`, which is already part of the dependency tree 

```suggestion
    "prepare": "is-ci || husky install",
```
we don't need `run-script-os`

**Review by sibiraj-s — CHANGES_REQUESTED** [review]
Looks good. Suggestion to replace `run-script-os` with `is-ci`

**Comment by paulirwin** [comment]
@sibiraj-s PR updated with those changes.

**Review by sibiraj-s — APPROVED** [review]

**Comment by sibiraj-s** [comment]
Sorry for the delay. had some personal emergencies.

**Pull request merged**